Trigger FrameRateMonitor updates on every frame

micuintus opened the pull request:
to obtain accurate frame rate measurements even when all items are inactive.

carlonluca commented:
Apparently you are making the component trigger (unneeded) repaints as fast as possible. This is not what a monitor should do though. The monitor should just display the frequency based on the app behavior. If the user wants to test how many repaints the system can do, he can do that externally.

micuintus commented:
I have now made this change optional for the user. The idea is this: I want to measure the performance of my `QQuickItem` and detect when the frame rate drops because my item is performing badly, not because of user inactivity. 

![fps](https://github.com/carlonluca/lqtutils/assets/8768660/97e1d40e-c899-44d1-8c3e-6c1d4e64b1df)

carlonluca commented:
I can understand the context, but I do not see the need as you can trigger the update externally from the class. A monitor typically observes, no reason to alter the behavior of the observed object.

However, as this is completely optional, if you still think this is useful, fine for me.

micuintus commented:
> A monitor typically observes

Fair enough... I see the point :) (separation of concerns).

I could also provide an extra object that triggers the update on every frame. (Could also be accomplished with a `QQuickPaintedItem` that calls `update()` on every paint BTW.) . 

> if you still think this is useful, fine for me.

OTOH, having this functionality as an option within the frame monitor and not having to use another item is very convenient.  🤷‍♂️ 

What do you think?

carlonluca commented:
For me, it is a go.

micuintus commented:
Then I'd vote let's merge this!

Nice set of tools on offer here BTW - thanks for that! I would even say they are cute :)

carlonluca commented:
Ok for me. I'll wait for the changes.

Commit:
Add helper function enableAutoFrameUpdates(...)

accompanying the FrameRateMonitor, which which causes the window to
request an update on every frame swap (that is on every vsync in most cases).

Commit:
Provide docu for lqt::enableAutoFrameUpdates()
